### ISSUE TITLE
Punctuation after doi and case of backref text

### ISSUE THREAD

**Issue by nkr0** [plk/biblatex-apa#165]
1) After a DOI link, `incollection`, `inbook`, or `book` type entries get a punctuation. Whereas, a regular article doesn't. In the screenshot, entries 2, 3, and 4 are journal articles.
2) Similarly, the `backref` text `Cit.` gets an uppercase `C` for the former type of entries and a lowercase `c` for articles.

![biblatex-apa](https://user-images.githubusercontent.com/42086312/164625735-4c70da13-1756-40d9-9e65-3cba2c470abc.png)

**Comment by moewew** [comment]
I'm guessing this is the same issue as https://github.com/plk/biblatex-apa/issues/141, which is resolved in the dev version of `biblatex-apa` here on GitHub (cf. https://github.com/plk/biblatex-apa/pull/159). If you can share the `.bib` file (or at least some of the relevant entries) that was used to produce the screenshot I could verify that or see if the fix needs additional work. (Note that sharing the relevant `.bib` entries and an example document that lets other people reproduce the output you are getting alongside the image is *always* appreciated. Often it's the only way to figure out what is going wrong.)

**Comment by nkr0** [comment]
The sample below will produce the 6 entries in the screenshot above. I had seen #141 before filing this, but I wasn't sure whether the bug in the DOI field was also included in that fix. Also, the issue with the case of backref text is new.

I am using the version from Debian testing and `apa.bbx` says `2021/12/24 v9.15`; which seems odd because the latest tag I see here is `9.14`. Anyway, #159 is from this Feb.

```
\documentclass[12pt,a4paper]{article}

\usepackage[style=apa, backref=true]{biblatex}

\begin{filecontents}[overwrite]{\jobname.bib}
@incollection{a,
  Author = {Ayhan Aksu-Koç and Dan Slobin},  
  title = {The Acquisition of {T}urkish},
  editor    = {Slobin, Dan},
  publisher = {Psychology Press},
  volume = {1},
  booktitle = {The Cross-Linguistic Study of Language Acquisition: The Data},
  year      = {1985},
  doi="10.4324/9781315802541"
}
@article{b,
  author = {Courtney B. Cazden},
  journal = {Child Development},
  number = {2},
  pages = {433–448},
  title = {The Acquisition of Noun and Verb Inflections},
  volume = {39},
  year = {1968},
  doi = {10.2307/1126956}
}
@article{c, 
  title={The development of regular and irregular verb inflection in Spanish child language}, 
  volume={29},
  DOI={10.1017/S0305000902005172}, 
  number={3}, 
  journal={Journal of Child Language}, 
  publisher={Cambridge University Press}, 
  author={Clahsen, Harald and Aveledo, Fraibet and Roca, Iggy}, 
  year={2002}, 
  pages={591–622}}
@article{d,
  title={Regular and irregular inflection in the acquisition of German noun plurals},
  author={Harald Clahsen and Monika Rothweiler and Andreas Woest and Gary F. Marcus},
  journal={Cognition},
  year={1992},
  volume={45},
  pages={225–255},
  doi={10.1016/0010-0277(92)90018-D}
}
@book{e,
  author    = "Clark, Eve V.",
  title     = "First Language Acquisition",
  publisher = "Cambridge University Press",
  year      = 2016,
  edition={3},
  doi ={10.1017/CBO9781316534175}
}
@incollection{f,
  Author = {Daugherty, K. and M. Seidenberg},
  title = "Beyond rules and exceptions: {A} connectionist approach to inflectional morphology",
  Editor = {S. D. Lima and R. L. Corrigan and G. K. Iverson},
  Pages = {353–388},
  Publisher = {John Benjamins Publishing Company},
  booktitle = {The reality of linguistic rules},
  series={Studies in language companion series 26},
  number={26},
  doi="10.1075/slcs.26.22dau",
  Year = {1994}
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}
\autocite{a,b,c,d,e,f}

\printbibliography
\end{document}
```

**Comment by moewew** [comment]
With the dev version (and the `dev` version of `biblatex` and Biber) the undesirable periods are all gone

> Aksu-Koç, A., & Slobin, D. (1985). The acquisition of Turkish. In D. Slobin (Ed.), The cross-linguistic study of language acquisition: The data (Vol. 1). Psychology Press. https://doi.org/10.4324/9781315802541 (cit. on p. 1).
> Cazden, C. B. (1968). The acquisition of noun and verb inflections. Child Development, 39(2), 433–448. https://doi.org/10.2307/1126956 (cit. on p. 1).
> Clahsen, H., Aveledo, F., & Roca, I. (2002). The development of regular and irregular verb inflection in spanish child language. Journal of Child Language, 29(3), 591–622. https://doi.org/10.1017/S0305000902005172 (cit. on p. 1).
> Clahsen, H., Rothweiler, M., Woest, A., & Marcus, G. F. (1992). Regular and irregular inflection in the acquisition of german noun plurals. Cognition, 45, 225–255. https://doi.org/10.1016/0010-0277(92)90018-D (cit. on p. 1).
> Clark, E. V. (2016). First language acquisition (3rd ed.). Cambridge University Press. https://doi.org/10.1017/CBO9781316534175 (cit. on p. 1).
> Daugherty, K., & Seidenberg, M. (1994). Beyond rules and exceptions: A connectionist approach to inflectional morphology. In S. D. Lima, R. L. Corrigan, & G. K. Iverson (Eds.), The reality of linguistic rules (pp. 353–388). John Benjamins Publishing Company. https://doi.org/10.1075/slcs.26.22dau (cit. on p. 1).

![screenshot of the output reproduced above](https://user-images.githubusercontent.com/6755835/164738372-76d6cd0e-2c8a-4d13-85a0-a0f7c213c55d.png)

So this was fixed by #159.

The different capitalisation of "cit on" was caused directly by the punctuation. The period was a sentnce-ending period after which `biblatex` capitalises automatically.

**Comment by nkr0** [comment]
Ok, thanks. I'll close this issue.
